### PR TITLE
feat(TDP-3412): logout

### PR DESCRIPTION
**Pull request by tlnd-jbrun**
**Link to the JIRA issue**
e.g. https://jira.talendforge.org/browse/TDP-3412

**Please check if the PR fulfills these requirements**
- [x] The commit(s) message(s) follows our [guidelines](https://github.com/talend/tools/blob/master/tools-root-github/CONTRIBUTING.md#commit-message-format)
- [x] Tests for the changes have been added (for bug fixes / features)
- [x] Docs have been added / updated (for bug fixes / features)
- [ ] The code coverage on new code is > 75 % for backend and > 95% for frontend
- [x] The new code does not introduce new technical issues (sonar / eslint)
- [x] Functional tests have been performed

**Please check the browsers you've tested on**
- [x] Chrome, Firefox, Safari, Edge, IE11
- [ ] No, that's bad, this PR should not be merged !
- [ ] No, and no need to (backend changes only)

**(Optional) What is the current behavior?**
(Additional information to the Jira)

**(Optional) What is the new behavior?**
(Additional information to the Jira)

**(Optional) Other information**:

### CODE REVIEW

**Comment by build-talend-tdp** [comment]
Pull request test result: [success](https://ci-tdp.datapwn.com/job/data-prep_pull-request_master/79/) [:star:](https://ci-tdp.datapwn.com/job/data-prep_pull-request_master/79/)

**Comment by build-talend-tdp** [comment]
Pull request test result: [success](https://ci-tdp.datapwn.com/job/data-prep_pull-request_master/80/) [:star:](https://ci-tdp.datapwn.com/job/data-prep_pull-request_master/80/)

**Comment by build-talend-tdp** [comment]
Pull request test result: [success](https://ci-tdp.datapwn.com/job/data-prep_pull-request_master/82/) [:star:](https://ci-tdp.datapwn.com/job/data-prep_pull-request_master/82/)

**Comment by build-talend-tdp** [comment]
Pull request test result: [success](https://ci-tdp.datapwn.com/job/data-prep_pull-request_master/83/) [:star:](https://ci-tdp.datapwn.com/job/data-prep_pull-request_master/83/)

File: dataprep-webapp/src/app/services/utils/config/utils-rest-urls-service.js [hunk 1]
```diff
@@ -24,20 +24,23 @@ export default function RestURLs() {
...
-		this.previewUrl = serverUrl + '/api/preparations/preview';
-		this.searchUrl = serverUrl + '/api/search';
-		this.settingsUrl = serverUrl + '/api/settings';
-		this.tcompUrl = serverUrl + '/api/tcomp';
-		this.transformUrl = serverUrl + '/api/transform';
-		this.typesUrl = serverUrl + '/api/types';
-		this.upgradeVersion = serverUrl + '/api/upgrade/check';
-		this.versionUrl = serverUrl + '/api/version';
+	this.setConfig = function setConfig(config, settings) {
+		const serverUrl = config.serverUrl;
```
Comment (frassinier) [inline dataprep-webapp/src/app/services/utils/config/utils-rest-urls-service.js]: Could be `const { serverUrl } = config;`

File: dataprep-webapp/src/app/services/utils/config/utils-rest-urls-service.js [hunk 2]
```diff
@@ -24,20 +24,23 @@ export default function RestURLs() {
...
+		const uris = settings.uris;
```
Comment (frassinier) [inline dataprep-webapp/src/app/services/utils/config/utils-rest-urls-service.js]: Could be `const { uris } = settings;`

**Comment by build-talend-tdp** [comment]
Pull request test result: [success](https://ci-tdp.datapwn.com/job/data-prep_pull-request_master/84/) [:star:](https://ci-tdp.datapwn.com/job/data-prep_pull-request_master/84/)

**Comment by build-talend-tdp** [comment]
Pull request test result: [success](https://ci-tdp.datapwn.com/job/data-prep_pull-request_master/88/) [:star:](https://ci-tdp.datapwn.com/job/data-prep_pull-request_master/88/)

**Comment by build-talend-tdp** [comment]
Pull request test result: [success](https://ci-tdp.datapwn.com/job/data-prep_pull-request_master/92/) [:star:](https://ci-tdp.datapwn.com/job/data-prep_pull-request_master/92/)

File: dataprep-api/src/main/java/org/talend/dataprep/api/service/settings/uris/api/UriSettings.java [hunk 3]
```diff
@@ -0,0 +1,95 @@
...
+
+    public void setUri(String uri) {
+        this.uri = uri;
+    }
+
+    /**
+     * Builder Pattern for immutable purpose
+     */
+
+    public static Builder from(final UriSettings actionSettings) {
```
Comment (jsomsanith) [inline dataprep-api/src/main/java/org/talend/dataprep/api/service/settings/uris/api/UriSettings.java]: This is not used --> to remove

File: dataprep-api/src/main/java/org/talend/dataprep/api/service/settings/AppSettingsService.java [hunk 4]
```diff
@@ -56,6 +63,10 @@ public AppSettings getSettings() {
         getSettingsStream(viewsProviders, viewsConfigurers) //
                 .forEach(view -> appSettings.getViews().put(view.getId(), view));
 
+        // populate appSettings uris (key: uriName, value: uriValue)
+        getSettingsStream(urisProviders, urisConfigurers) //
+                .forEach(uri -> appSettings.getUris().put(uri.getName(), uri.getUri()));
```
Comment (jsomsanith) [inline dataprep-api/src/main/java/org/talend/dataprep/api/service/settings/AppSettingsService.java]: Can you align the name of uri `name` to uri `id` ? (like the other settings entities actions and views which have ids to identify them)

File: dataprep-api/src/test/java/org/talend/dataprep/api/service/settings/UrisAppSettingsAPITest.java [hunk 5]
```diff
@@ -0,0 +1,93 @@
...
+package org.talend.dataprep.api.service.settings;
+
+import static com.jayway.restassured.RestAssured.when;
+import static org.hamcrest.Matchers.is;
+import static org.junit.Assert.assertThat;
+
+import java.util.Map;
+
+import org.junit.Test;
+import org.springframework.test.context.TestPropertySource;
```
Comment (jsomsanith) [inline dataprep-api/src/test/java/org/talend/dataprep/api/service/settings/UrisAppSettingsAPITest.java]: Unused import

File: dataprep-api/src/test/java/org/talend/dataprep/api/service/settings/UrisAppSettingsAPITest.java [hunk 6]
```diff
@@ -0,0 +1,93 @@
...
+import org.talend.dataprep.api.service.ApiServiceTestBase;
+
+public class UrisAppSettingsAPITest extends ApiServiceTestBase {
```
Comment (jsomsanith) [inline dataprep-api/src/test/java/org/talend/dataprep/api/service/settings/UrisAppSettingsAPITest.java]: Shouldn't this test be in AppSettingsAPITest ?

File: dataprep-api/src/main/java/org/talend/dataprep/api/service/settings/uris/api/UriSettings.java (hunk 3 above)
Comment (tlnd-jbrun) [inline dataprep-api/src/main/java/org/talend/dataprep/api/service/settings/uris/api/UriSettings.java]: done

File: dataprep-api/src/main/java/org/talend/dataprep/api/service/settings/AppSettingsService.java (hunk 4 above)
Comment (tlnd-jbrun) [inline dataprep-api/src/main/java/org/talend/dataprep/api/service/settings/AppSettingsService.java]: done

File: dataprep-api/src/test/java/org/talend/dataprep/api/service/settings/UrisAppSettingsAPITest.java (hunk 5 above)
Comment (tlnd-jbrun) [inline dataprep-api/src/test/java/org/talend/dataprep/api/service/settings/UrisAppSettingsAPITest.java]: done

File: dataprep-api/src/test/java/org/talend/dataprep/api/service/settings/UrisAppSettingsAPITest.java (hunk 6 above)
Comment (tlnd-jbrun) [inline dataprep-api/src/test/java/org/talend/dataprep/api/service/settings/UrisAppSettingsAPITest.java]: done

**Comment by build-talend-tdp** [comment]
Pull request test result: [success](https://ci-tdp.datapwn.com/job/data-prep_pull-request_master/94/) [:star:](https://ci-tdp.datapwn.com/job/data-prep_pull-request_master/94/)

**Comment by build-talend-tdp** [comment]
Pull request test result: [success](https://ci-tdp.datapwn.com/job/data-prep_pull-request_master/98/) [:star:](https://ci-tdp.datapwn.com/job/data-prep_pull-request_master/98/)

File: dataprep-webapp/src/app/index-module.js [hunk 7]
```diff
@@ -84,16 +84,16 @@ window.fetchConfiguration = function fetchConfiguration() {
 					'ngInject';
 					$compileProvider.debugInfoEnabled(config.enableDebug);
 				})
-				// Configure server api urls
-				.run((RestURLs) => {
-					'ngInject';
-					RestURLs.setConfig(config);
-				})
 				// Fetch dynamic configuration (supported encodings, ...)
```
Comment (ncomont) [inline dataprep-webapp/src/app/index-module.js]: done, thanks

**Comment by build-talend-tdp** [comment]
Pull request test result: [success](https://ci-tdp.datapwn.com/job/data-prep_pull-request_master/106/) [:star:](https://ci-tdp.datapwn.com/job/data-prep_pull-request_master/106/)

**Comment by build-talend-tdp** [comment]
Pull request test result: [success](https://ci-tdp.datapwn.com/job/data-prep_pull-request_master/107/) [:star:](https://ci-tdp.datapwn.com/job/data-prep_pull-request_master/107/)

**Comment by build-talend-tdp** [comment]
Pull request test result: [failure](https://ci-tdp.datapwn.com/job/data-prep_pull-request_master/124/) [:x:](https://ci-tdp.datapwn.com/job/data-prep_pull-request_master/124/)

**Comment by build-talend-tdp** [comment]
Pull request test result: [success](https://ci-tdp.datapwn.com/job/data-prep_pull-request_master/129/) [:star:](https://ci-tdp.datapwn.com/job/data-prep_pull-request_master/129/)

File: dataprep-api/src/main/java/org/talend/dataprep/api/service/settings/uris/api/UriSettings.java (hunk 3 above)
Comment (jsomsanith) [inline dataprep-api/src/main/java/org/talend/dataprep/api/service/settings/uris/api/UriSettings.java]: Still present but not used :)

**Pull request merged**